onnxruntime/core/mlas/inc/mlas_qnbit.h

```diff
@@ -0,0 +1,79 @@
+/*++
+
+Copyright (c) Microsoft Corporation. All rights reserved.
+
+Licensed under the MIT License.
+
+Module Name:
+
+    mlas_qnbit.h
+
+Abstract:
+
+    This module contains the public data structures and procedure prototypes
+    for blocked n-bit quantized GEMM.
+
+    N-bit block quantization is used to compress weight tensors of large
+    language models.
+
+--*/
+
+#pragma once
+
+#include "mlas.h"
+#include "mlas_gemm_postprocessor.h"
+
+/**
+ * @brief Data parameters for float/n-bit quantized int GEMM routine.
+ */
+struct MLAS_SQNBIT_GEMM_DATA_PARAMS {
+    const float* A = nullptr;                ///< address of A (float32 matrix)
+    const void* PackedBData = nullptr;       ///< address of B (quantized and packed n-bit int values)
```

Comment:
Is BData, BScale and BZeroPoint expected to be packed? I thought they are not packed and only pre-packed if IsBPacked is true.

Reply:
maybe we can find a better name. I'm actually not too clear on the definition of "packed". currently IsBPacked is unused.

but to define these parameters:

assuming 4-bit int quantization,
BlockCountK = ceil(K / BlkLen)

PackedBData: [N][BlockCountK][BlkLen] 4-bit int data values
PackedBScale: [N][BlockCountK] float scale values
PackedBZeroPoint: [N][BlockCountK] 4-bit int zero point values. (note that it is [N][ceil(BlockCountK / 2)] bytes to account for a padding half byte if BlockCountK is odd).

Reply:
renamed to QuantBData, QuantBScale, QuantBZeroPoint